curriculum/challenges/english/blocks/lecture-working-with-npm-scripts/692aba0a78ca391f10455b4b.md

```diff
@@ -0,0 +1,249 @@
+---
+id: 692aba0a78ca391f10455b4b
+title: How Do Imports and Exports Work with CommonJS Modules?
+challengeType: 19
+dashedName: how-do-imports-and-exports-work-with-commonjs-modules
+---
+
+# --description--
+
+In modern web development, you can split your code into multiple files called "modules."
+
+You can think of a module as a self-contained and reusable code block.
+
+By importing modules into other modules, you can reuse their functionality while still keeping the code in separate files, which is more maintainable, especially for large and complex web applications.
+
+In Node.js, each file is considered a module.
+
+Node has two different systems for working with modules: CommonJS and ES.
+
+CommonJS is the original and default way of working with packages in Node.js. It loads modules synchronously, which means that it has to wait for a module to be fully loaded before it start loading the next one or continues running the code.
+
+It's still widely used, but it's gradually being replaced by the newer ES syntax.
+
+CommonJS uses the `require()` function to import a module, taking its relative path as argument.
+
+For example, let's say that we have a module called `math.js`, where we define basic mathematical operations.
+
+We'll only use a multiplication function here to keep this example simple and focused.
```

Comment:
````suggestion
We'll only use a multiplication function here to keep this example simple and focused:

```js
function multiply(a, b) {
  return a * b;
}
```
````

I might add the example here.